.github/workflows/cd-pipeline.yaml

```diff
@@ -0,0 +1,93 @@
+            commits=($(echo "${GITHUB_COMMITS}" | jq -r '.[] | .id'))
+            flavors=()
+            for commit in "${commits[@]}"; do
+              # Using the Github API to detect the files changed as git merge-base stops working when the branch is behind
+              URL="https://api.github.com/repos/${{ github.repository }}/commits/${commit}"
+              files_changed_data=$(curl -s --header 'authorization: Bearer ${{ secrets.GITHUB_TOKEN }}' -X GET -G "$URL")
+              files_changed="$(echo $files_changed_data | jq -r '.files[] | .filename')"
+              # Adding || true to avoid "Process exited with code 1" errors
+              flavors+=($(echo "$files_changed" | xargs dirname | grep -o "containers/[^/]*/[^/]*/[^/]*" | sort | uniq || true))
+            done
+
+            if [[ "${#flavors[@]}" -le "0" ]]; then
+              echo "::set-output name=error::No changes detected in containers. The rest of the tests will be skipped."
+              echo "::set-output name=result::skip"
+            else
+              flavors_json=$(printf "%s\n" "${flavors[@]}" | jq -R . | jq -cs .)
+              echo "::set-output name=result::ok"
+              echo "::set-output name=flavors::${flavors_json}"
+            fi
+          else
+            echo "::set-output name=result::skip"
+            echo "::set-output name=error::Forced push detected. CD pipeline will be skipped"
+          fi
+      # Using actions/github-scripts because using exit 1 in the script above would not provide any output
+      # Source: https://github.community/t/no-output-on-process-completed-with-exit-code-1/123821/3
+      - id: show-error
+        name: Show error
+        if: ${{ steps.get-container.outputs.result != 'ok' }}
+        uses: actions/github-script@v3
+        with:
+          script: |
+              core.setFailed('${{ steps.get-container.outputs.error }}')
+  vib-publish:
+    runs-on: ubuntu-latest
+    needs: get-container
+    if: ${{ needs.get-container.outputs.result == 'ok' }}
+    name: publish
+    strategy:
+      fail-fast: false
+      matrix:
+        flavor: ${{ fromJSON(needs.get-container.outputs.flavors) }}
+    steps:
+      - uses: actions/checkout@v3
+        name: Checkout Repository
+        # Required to search the latest commit with the tag
+        with:
+          fetch-depth: 0
+      - id: get-metadata
+        name: Get image tag and container name
+        run: |
+          tag="$(git log --pretty=tformat:"%s" -n 1 --grep=" release$" --author bitnami-bot@vmware.com --author containers@bitnami.com --author containers-bot@bitnami.com --  ${{ matrix.flavor }} | awk '{print $1}')"
+          container="$(echo "${{matrix.flavor}}" | awk -F '/' '{print $2}')"
+          echo "::set-output name=tag::${tag}"
+          echo "::set-output name=container::${container}"
+      - uses: vmware-labs/vmware-image-builder-action@main
+        name: Publish ${{ needs.get-container.outputs.container }}
```

Comment:
`get-container` does no have `container` as declared outputs (which indeed is strange for a job named like that), only `result` and `flavors`. Maybe you meant `get-metadata` instead?